File: apzkr-pzpi-21-11-milennyi-ivan/Task3-WebClient/src/models/ShearingLog.ts
// Клас, що представляє запис стрижки
export class ShearingLog {
    id: number | null; // Ідентифікатор запису стрижки
    sheep_id: number; // Ідентифікатор вівці
    shepherd_id: number | null; // Ідентифікатор пастуха
    timestamp: number; // Час стрижки у секундах
    wool_amount: number; // Кількість вовни

    constructor(id: number | null, sheepId: number, shepherdId: number | null, timestamp: number, woolAmount: number) {
        this.id = id;
        this.sheep_id = sheepId;
        this.shepherd_id = shepherdId;
        this.timestamp = Math.floor(timestamp / 1000);
        this.wool_amount = woolAmount;
    }
}

File: apzkr-pzpi-21-11-milennyi-ivan/Task1-Server/src/models/shearing_log.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use utoipa::ToSchema;
use validator::Validate;

// Структура, що представляє журнал стрижки
#[derive(Debug, Validate, Deserialize, Serialize, FromRow, ToSchema)]
pub(crate) struct ShearingLog {
    id: Option<u64>, // Ідентифікатор логу стрижки
    sheep_id: u64, // Ідентифікатор вівці
    shepherd_id: Option<u64>, // Ідентифікатор пастуха
    timestamp: u64, // Час стрижки (у вигляді таймстемпу)
    #[validate(range(min = 1))]
    wool_amount: u32 // Кількість вовни
}

impl ShearingLog {
    // Отримання ідентифікатора логу стрижки
    pub fn id(&self) -> Option<u64> {
        self.id
    }

    // Встановлення ідентифікатора логу стрижки
    pub fn set_id(&mut self, id: u64) -> () {
        self.id = Some(id);
    }

    // Отримання ідентифікатора вівці
    pub fn sheep_id(&self) -> u64 {
        self.sheep_id
    }

    // Отримання ідентифікатора пастуха
    pub fn shepherd_id(&self) -> Option<u64> {
        self.shepherd_id
    }

    // Отримання часу стрижки
    pub fn timestamp(&self) -> u64 {
        self.timestamp
    }

    // Отримання кількості вовни
    pub fn wool_amount(&self) -> u32 {
        self.wool_amount
    }
}
